Helper/Cart.php

```diff
@@ -1334,19 +1443,20 @@ public function getCartData($paymentOnly, $placeOrderPayload, $immutableQuote =
                 $balanceModel->setCustomer(
                     $this->customerSession->getCustomer()
                 )->setWebsiteId(
-                    $this->checkoutSession->getQuote()->getStore()->getWebsiteId()
+                    $this->getWebsiteId()
                 );
                 $balanceModel->loadByCustomer();
 
                 if ($amount = abs($balanceModel->getAmount())) {
                     $roundedAmount = $this->getRoundAmount($amount);
 
-                    $cart['discounts'][] = [
+                    $discounts[] = [
                         'description' => 'Store Credit',
                         'amount'      => $roundedAmount,
                         'type'        => 'fixed_amount',
                     ];
 
+                    $diff -= $amount * 100 - $roundedAmount;
```

Comment:
Also, i would be glad if there appears unit tests which covered different variable types here (e.x `float`, `int` and etc).

CC: @danac-gs @matt-thomason

Reply:
No need for $amount and $roundedAmount to be of the same type. $roundedAmount is float with no decimal part, ~ integer. Will be explicitly casted to int per request above. @amount is some number, int or float. $diff can be 0 or some float. Later in the code, only if the accumulated $diff is over some threshold we take it into account. We don't check for equality so floating point precision can't get in our way.

Reply:
@danac-gs  if we already have `$roundedAmount` as `int`.
Does it make sense to convert `$amount` to `int` also.
i mean something like that: `$diff -= (int) $amount * 100 - $roundedAmount;`

What the type return `$balanceModel->getAmount()` ?

As i remember correct about `abs()` we should have the following:
1. `$diff -= abs(12.3)*100 - 100;`
The result would be `float(1130)`
2. `$diff -= abs(123)*100 - 100;`
The result would be `int(12200)`

I do not see a problem here, and i am ok with such implementation, but if we work only on `int` values like `100` and `2235` without comma, i think it would be better to use only `int` types.

What do you think? ;-)